python/ray/data/_internal/arrow_ops/transform_pyarrow.py

```
@@ -179,6 +179,14 @@ def unify_schemas(
         ArrowVariableShapedTensorType,
     )
 
+    try:
+        if len(set(schemas)) == 1:
+            # Early exit because unifying can be expensive
+            return schemas[0]
+    except Exception:
+        # Unsure if there are cases where schemas are NOT hashable
+        pass
```

Don't swallow exceptions